crates/rnote-ui/src/appwindow/mod.rs

```diff
@@ -104,6 +104,30 @@ impl RnAppWindow {
         self.imp().overlays.get()
     }
 
+    pub(crate) fn get_number_of_tabs(&self) -> u32 {
+        self.imp().overlays.tabview().pages().n_items()
+    }
+
+    pub(crate) fn get_tab_i(&self, i: u32) -> Result<RnCanvas, anyhow::Error> {
+        let glib_obj = self.imp().overlays.tabview().pages().item(i);
+        if !glib_obj.is_some() {
+            return Err(anyhow::anyhow!("could not obtain the tab {:?}", i));
+        }
+        let tab_glib = glib_obj.unwrap().downcast::<adw::TabPage>();
```

Comment:
It is fine to unwrap (or `expect`) here, the object downcasting to RnCanvasWrapper always succeeds. It is done in other places in the code as well.

Reply:
fixed